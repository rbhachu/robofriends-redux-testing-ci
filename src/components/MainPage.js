import React, { Component } from 'react';
import CardsList from './CardsList';
import SearchBox from './SearchBox';
import ErrorBoundary from './ErrorBoundary';

export class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots();
  }
  
  render() {
    const { robots, searchField, onSearchChange, isPending } = this.props; //get props
    const filteredRobots = robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
  })

    return (
      <>
          { isPending ? 
              <div className='tc'>
                  <header>
                      <h1 className='f1 fw2 light-blue'>Loading...</h1>
                  </header>
              </div>
          :
              <div className='tc'>
                  <header>
                      <h1 className='f1 fw2 light-blue'>RoboFriends</h1>
                  </header>
                  <section>
                      <SearchBox searchChange={onSearchChange}/>
                      <ErrorBoundary>
                      <CardsList robots={filteredRobots} /> 
                      </ErrorBoundary>
                  </section>
              </div>
          }
      </>
    )

  }
}

export default MainPage