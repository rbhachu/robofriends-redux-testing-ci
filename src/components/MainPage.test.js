import React from 'react';
import { Provider } from 'react-redux'; // import statement for react redux
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps}/>)
 })

it('renders without crashing', () => { // test name
  expect(wrapper).toMatchSnapshot();
});

it('filters Robots', () => { // test name
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: 'a',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps}/>)
  expect(wrapper).toMatchSnapshot();
  //expect(wrapper.instance().filteredRobots()).toEqual([]);
});

it('filters Robots correctly', () => { // test name
  const filteredRobots = [{
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz'
  }]
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz'
    }],
    searchField: 'Leanne',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps}/>)
  expect(wrapper).toMatchSnapshot();
  //expect(wrapper.instance().filteredRobots()).toEqual(filteredRobots);
});

it('filters Robots correctly 2', () => { // test name
  const filteredRobots = [{
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz'
  }]
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz'
    }],
    searchField: 'Xavier',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps}/>)
  expect(wrapper).toMatchSnapshot();
  //expect(wrapper.instance().filteredRobots()).toEqual([]);
});

