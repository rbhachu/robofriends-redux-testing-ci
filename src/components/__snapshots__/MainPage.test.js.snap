// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters Robots 1`] = `ShallowWrapper {}`;

exports[`filters Robots correctly 1`] = `ShallowWrapper {}`;

exports[`filters Robots correctly 2 1`] = `ShallowWrapper {}`;

exports[`renders without crashing 1`] = `ShallowWrapper {}`;
